### PR TITLE
Add pre-commit session & add default sessions

### PR DESCRIPTION
**Pull request by arjxn-py**
# Description

This PR aims to add a pre-commit nox session.

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all`
- [ ] The documentation builds: `$ python run-tests.py --doctest`

You can run unit and doctests together at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by arjxn-py** [comment]
@Saransh-cpp do I need to add this session to CI somewhere as well?

**Comment by agriyakhetarpal** [comment]
We already have a `pre-commit` [bot](https://github.com/apps/pre-commit-ci) and a [style check](https://github.com/pybamm-team/PyBaMM/actions/runs/5417104836/workflow?pr=3084#L33-L36) job in the CI

**Comment by arjxn-py** [comment]
> We already have a `pre-commit` [bot](https://github.com/apps/pre-commit-ci) and a [style check](https://github.com/pybamm-team/PyBaMM/actions/runs/5417104836/workflow?pr=3084#L33-L36) job in the CI

Do we want to execute this via `nox` as @Saransh-cpp had listed adding a pre-commit session in [#3048(comment)](https://github.com/pybamm-team/PyBaMM/issues/3048#issue-1762366935)

**Comment by agriyakhetarpal** [comment]
I think it would cause an unnecessary intermediate step plus some overhead in the CI, but it's good for executing locally if we list it as a [default session](https://nox.thea.codes/en/stable/config.html#modifying-nox-s-behavior-in-the-noxfile) – running `nox` would mean the same as running `nox -s pre-commit`

**Comment by Saransh-cpp** [comment]
> we list it as a [default session](https://nox.thea.codes/en/stable/config.html#modifying-nox-s-behavior-in-the-noxfile)

That is exactly what I had in mind. We can add the unit session and the pre-commit session as the default ones.

**Comment by arjxn-py** [comment]
@Saransh-cpp I've added pre-commit & unit in default sessions.

File: noxfile.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 import os
 import sys
 
+nox.options.sessions = ["pre-commit", "unit"]
+
```
[the remark this answers is not in the document]
Comment (arjxn-py) [reply noxfile.py]: I did that, didn't felt modifying documentation much for this so modified `nox -s unit` section a bit. Please let me know if more is required.

[the remark this answers is not in the document]
Comment (Saransh-cpp) [reply noxfile.py]: This should actually go in the place of `nox -s tests` above this, with a minor modification in the test. Maybe we can have a table like structure for nox commands, just like we have for optional dependencies.

**Review by Saransh-cpp — APPROVED** [review]

**Comment by Saransh-cpp** [comment]
Nice, thanks, @arjxn-py!

**Pull request merged**